src/components/textbox/textbox.pw.tsx

```diff
@@ -0,0 +1,1127 @@
+        mount,
+        page,
+      }) => {
+        await mount(<TextboxComponent labelInline labelSpacing={spacing} />);
+
+        const label = getDataElementByValue(page, "label");
+        const labelParent = await label.evaluateHandle((element: Element) => {
+          return element.parentElement;
+        });
+
+        const paddingRightValue = await page.evaluate(
+          (element) =>
+            window.getComputedStyle(element).getPropertyValue("padding-right"),
+          labelParent
+        );
+        expect(paddingRightValue).toEqual(padding);
+      });
+    }
+  );
+
+  ([
+    [10, 90, 135, 1229],
+    [30, 70, 409, 956],
+    [80, 20, 1092, 273],
+  ] as [
+    TextboxProps["labelWidth"],
+    TextboxProps["inputWidth"],
+    number,
+    number
+  ][]).forEach(([labelVal, inputVal, labelRatio, inputRatio]) => {
+    test(`should render with correct ratios when inputWidth prop is ${inputVal} and labelWidth prop is ${labelVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <TextboxComponent
+          labelInline
+          labelWidth={labelVal}
+          inputWidth={inputVal}
+        />
+      );
+
+      const label = getDataElementByValue(page, "label");
+      const labelParent = await label.evaluateHandle((element: Element) => {
+        return element.parentElement;
+      });
+      const textboxJsHandle = await textbox(page).evaluateHandle(
+        (element: Element) => {
+          return element;
+        }
+      );
+
+      const labelRatioValue = await page.evaluate(
+        (element) => window.getComputedStyle(element).getPropertyValue("width"),
+        labelParent
+      );
+      const inputRatioValue = await page.evaluate(
+        (element) => window.getComputedStyle(element).getPropertyValue("width"),
+        textboxJsHandle
+      );
+
+      expect(parseInt(labelRatioValue)).toBeGreaterThanOrEqual(
+        labelRatio - 0.5
+      );
+      expect(parseInt(inputRatioValue)).toBeLessThanOrEqual(inputRatio + 0.5);
+    });
+  });
+
+  [
+    ["foo", "with"],
+    ["", "without"],
+  ].forEach(([hint, renderStatus]) => {
+    test(`should render ${renderStatus} an input hint depending on the inputHint prop value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent inputHint={hint} />);
+
+      const inputHint = getDataElementByValue(page, "input-hint");
+
+      if (renderStatus === "with") {
+        await expect(inputHint).toHaveCount(1);
+      } else {
+        await expect(inputHint).toHaveCount(0);
+      }
+    });
+  });
+
+  ([
+    [4, "with", "is"],
+    ["", "without", "is not"],
+  ] as [TextboxProps["characterLimit"], string, string][]).forEach(
+    ([characterLimitVal, renderStatus, characterLimitStatus]) => {
+      test(`should render ${renderStatus} a character count hint when the characterLimit prop ${characterLimitStatus} passed`, async ({
```

Comment:
There were three cypress tests for this prop, but only one in playwright. Have the others been removed because they're unnecessary?

Reply:
Good spot, I actually added these in during my character count work which was in progress during this refactor so they've been missed. I'll add them in in a seperate commit